File: Template Method/Ruby/TwoFactorPhone.rb
require_relative "TwoFactorAuthentication"

class TwoFactorPhone < TwoFactorAuthentication
    def getData(username)
        puts "Getting phone data for #{username}"
    end

    def sendNotification(username)
        puts "Sending sms to #{username}"
    end

    def match(username)
        puts "Matching #{username} with the sms code"
    end
end

File: Template Method/Dart/TwoFactorPhone.dart
import 'TwoFactorAuthentication.dart';

class TwoFactorPhone extends TwoFactorAuthentication {
  @override
  void getData(String username) {
    print("Getting phone data for $username");
  }

  @override
  void match(String username) {
    print("Matching sms code from $username");
  }

  @override
  void sendNotification(String username) {
    print("Sending sms to $username");
  }
}
